src/dripper/polkadot/PolkadotActions.ts

```diff
@@ -1,223 +1,165 @@
+  async transferTokens(dripAmount: bigint, address: string): Promise<DripResponse> {
+    logger.info(`💸 sending tokens to ${address}`);
+    const tx = await networkData.api.getTransferTokensTx({
+      dripAmount,
+      address,
+      client,
+      nonce: this.#nonce++,
+    });
+    logger.debug(`Dripping to ${address}. Transaction ${tx}`);
+    // client.submit(tx) waits for the finalized value, which might be important for real money,
+    // but for the faucet drips, early respond is better UX
+
+    const submit$ = client.submitAndWatch(tx);
+    submit$.forEach((data) => {
+      logger.info(`submit$ ${address}`, data.type);
+    });
+
+    const hash = (await firstValueFrom(submit$.pipe(filter((value) => value.type === "broadcasted")))).txHash;
+
+    await firstValueFrom(submit$.pipe(filter((value) => value.type === "txBestBlocksState" && value.found)));
+
+    logger.info(`💸 sending tokens to ${address}: done: ${hash}`);
+    return { hash };
+  }
+
+  async sendTokens(address: string, parachain_id: number | null, amount: bigint): Promise<DripResponse> {
     let dripTimeout: ReturnType<typeof rpcTimeout> | null = null;
     let result: DripResponse;
-    const faucetBalance = this.getFaucetBalance();
 
     try {
-      if (!this.account) throw new Error("account not ready");
-
-      if (typeof faucetBalance !== "undefined" && amount >= faucetBalance) {
+      if (typeof this.#faucetBalance !== "undefined" && amount >= this.#faucetBalance) {
         const formattedAmount = formatAmount(amount);
-        const formattedBalance = formatAmount(faucetBalance);
+        const formattedBalance = formatAmount(this.#faucetBalance);
 
         throw new Error(
-          `Can't send ${formattedAmount} ${networkData.currency}s, as balance is only ${formattedBalance} ${networkData.currency}s.`,
+          `Can't send ${formattedAmount} ${networkData.data.currency}s, as balance is only ${formattedBalance} ${networkData.data.currency}s.`,
         );
       }
 
       // start a counter and log a timeout error if we didn't get an answer in time
       dripTimeout = rpcTimeout("drip");
-      if (parachain_id != "") {
+      if (parachain_id !== null) {
         result = await this.teleportTokens(amount, address, parachain_id);
       } else {
-        logger.info("💸 sending tokens");
-        const transfer = polkadotApi.tx.balances.transferKeepAlive(address, amount);
-        const hash = await transfer.signAndSend(this.account, { nonce: -1 });
-        result = { hash: hash.toHex() };
+        result = await this.transferTokens(amount, address);
       }
     } catch (e) {
-      result = { error: (e as Error).message || "An error occured when sending tokens" };
       logger.error("⭕ An error occured when sending tokens", e);
+      let message = "An error occured when sending tokens";
+      if (e instanceof Error) {
+        message = e.message;
+      }
```

Comment:
I never thought of re assigning the message object. I'll copy this in my code.

Reply:
It's just reassigning a string variable. You sure you saw it the right way?

Reply:
I was always doing something like:
```ts
if (e instanceof Error) {
    return e.message;
} else if (e instanceof CustomError) {
    return e.customMessage;
}
return "There was an error"
```

Now that I think about it's kind of the same code 😅 

Still, I hate error objects uncertainty in JS